torch/optim/sgd.py

```diff
@@ -100,7 +101,8 @@ def __init__(self, params, lr=required, momentum=0, dampening=0,
 
         defaults = dict(lr=lr, momentum=momentum, dampening=dampening,
                         weight_decay=weight_decay, nesterov=nesterov,
-                        maximize=maximize, foreach=foreach)
+                        maximize=maximize, foreach=foreach,
+                        differentiable=differentiable)
```

Comment:
Do we need `group.setdefault('differentiable', False)` in `__setstate__` below as well?

Reply:
What about we add this field directly to the base `Optimizer` class (in the `__setstate__` in particular).
We want this to be there in the end anyways and that will avoid any BC issue where we forget to add it for a specific optimizer.

Reply:
The `__setstate__` is only invoked when `load_state_dict` is called. How to add a `self._differentiable` to the `Optimizer.__init__`?